src/main/java/ui/components/pickers/LabelPickerDialog.java

```diff
@@ -202,25 +281,73 @@ private Tooltip createTitleTooltip(TurboIssue issue) {
         return titleTooltip;
     }
 
-    private FlowPane createTopPane() {
-        FlowPane topPane = new FlowPane();
-        topPane.setPadding(new Insets(20, 0, 10, 0));
-        topPane.setHgap(5);
-        topPane.setVgap(5);
-        return topPane;
+    private void setTitleLabel(TurboIssue issue) {
+        title.setText((issue.isPullRequest() ? "PR #" : "Issue #")
+                + issue.getId() + ": " + issue.getTitle());
+    }
+
+    private Label createTextLabel(String input) {
+        Label label = new Label(input);
+        label.setPadding(new Insets(2, 5, 2, 5));
+        return label;
     }
 
-    private TextField createTextField() {
-        TextField textField = new TextField();
-        textField.setId("labelPickerTextField");
-        textField.setPrefColumnCount(30);
-        return textField;
+
+    private Label createGroupTitle(String name) {
+        Label groupName = new Label(name);
+        groupName.setPadding(new Insets(0, 5, 5, 0));
+        groupName.setMaxWidth(ELEMENT_MAX_WIDTH - 10);
+        groupName.setStyle("-fx-font-size: 110%; -fx-font-weight: bold; ");
+        return groupName;
+    }
+
+    private FlowPane createGroupPane(Insets padding) {
+        FlowPane group = new FlowPane();
+        group.setHgap(5);
+        group.setVgap(5);
+        group.setPadding(padding);
+        return group;
+    }
+
+    // Event handling 
+
+    private void setupEvents(Stage stage) {
+        showingProperty().addListener(e -> {
+            positionDialog(stage);
+        });
+    }
+
+    @SuppressWarnings("PMD")
+    private void handleUserInput(ObservableValue<? extends String> observable, 
+                                 String oldText, String newText) {
+        state = new LabelPickerState(
+            new HashSet(issue.getLabels()), repoLabels, queryField.getText().toLowerCase());
+        populatePanes(state);
+    }
+
+    private void handleLabelClick(PickerLabel label) {
+        state.updateAssignedLabels(label);
+        populatePanes(state);
+
+        if (!queryField.isDisabled()) {
+            queryField.setDisable(true);
+        }
+        queryField.textProperty().removeListener(listener);
+        queryField.clear();
+    }
+
+    private void positionDialog(Stage stage) {
+        if (getDialogHeight().isPresent()) {
+            setX(stage.getX() + stage.getScene().getX());
+            setY(stage.getY() +
+                    stage.getScene().getY() +
+                    (stage.getScene().getHeight() - getHeight()) / 2);
+        }
     }
 
-    private VBox createBottomBox() {
-        VBox bottomBox = new VBox();
-        bottomBox.setPadding(new Insets(10, 0, 0, 0));
-        return bottomBox;
+    // getHeight() returns NaN when the height is not set
```

Comment:
Should be a javadoc comment. When is the height not set? Should it be set by whatever is creating the dialog?